### PR TITLE
Reserve host memory regardless address translator presence

### PR DESCRIPTION
**Pull request by chienwei-lan**

### CODE REVIEW

**Review by xuhz — APPROVED** [review]

**Review by houlz0507 — APPROVED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 1]
```diff
@@ -1278,14 +1273,12 @@ static int xocl_cma_mem_alloc_huge_page(struct xocl_dev *xdev, struct drm_xocl_a
 {
 	int ret = 0;
 	size_t page_sz = cma_info->total_size/cma_info->entry_num;
-	uint32_t i, j, num = xocl_addr_translator_get_entries_num(xdev);
+	uint32_t i, j, num = 256;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:1276]: please add a #define with some comments for this magic number "256"

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 2]
```diff
@@ -1455,14 +1452,9 @@ static int xocl_cma_mem_alloc(struct xocl_dev *xdev, uint64_t size)
 	int ret = 0;
 	uint64_t page_sz;
 	int64_t i = 0;
-	uint64_t page_num = xocl_addr_translator_get_entries_num(xdev);
+	uint64_t page_num = 256;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:1455]: use the #define above

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 3]
```diff
@@ -1534,12 +1527,7 @@ void xocl_cma_bank_free(struct xocl_dev	*xdev)
 int xocl_cma_bank_alloc(struct xocl_dev	*xdev, struct drm_xocl_alloc_cma_info *cma_info)
 {
 	int err = 0;
-	int num = xocl_addr_translator_get_entries_num(xdev);
-
-	if (!num) {
-		DRM_ERROR("Doesn't support HOST MEM feature");
-		return -ENODEV;
-	}
+	int num = 256;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:1530]: is this "256" the same as other "256"? if so, pls use the same macro.

**Review by maxzhen — CHANGES_REQUESTED** [review]

**Review by xuhz — COMMENTED** [review]
BTW, i tested the changes on a u50 nodma shell. 
With the other changes i made, moving m2m and SB from shell to ULP works -- i can run 'xbutil validate' successfully

**Review by maxzhen — APPROVED** [review]

**Pull request merged**